nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java

```diff
@@ -303,6 +303,9 @@ public class NiFiProperties extends ApplicationProperties {
     // flow analysis properties
     public static final String BACKGROUND_FLOW_ANALYSIS_SCHEDULE = "nifi.flow.analysis.background.task.schedule";
 
+    // registry client properties
+    public static final String FLOW_REGISTRY_CHECK_FOR_RULE_VIOLATIONS_BEFORE_COMMIT = "nifi.registry.check.for.rule.violation.before.commit";
```

Comment:
We should add this property in `nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/nifi.properties` and in `nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/pom.xml` as well.

 ```suggestion
    public static final String FLOW_REGISTRY_CHECK_FOR_RULE_VIOLATIONS_BEFORE_COMMIT = "nifi.registry.check.for.rule.violations.before.commit";
```